src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -60,77 +68,42 @@ def __call__(self, t):
         :obj:`RandomVariable`
             Estimate of the states at time ``t``.
         """
+
+        # Recursive evaluation (t can now be any array, not just length 1)
         if not np.isscalar(t):
-            # recursive evaluation (t can now be any array, not just length 1!)
             return _RandomVariableList([self.__call__(t_pt) for t_pt in np.asarray(t)])
 
+        # t is left of our grid -- raise error
+        # (this functionality is not supported yet)
         if t < self.locations[0]:
             raise ValueError(
                 "Invalid location; Can not compute posterior for a location earlier "
                 "than the initial location"
             )
 
+        # Early exit if t is in our grid -- no need to interpolate
         if t in self.locations:
-            idx = (self.locations <= t).sum() - 1
+            idx = self._find_previous_or_same_index(t)
```

Comment:
Do we actually need "previous or same" here? Since we have `t in self.locations` we're just looking for the index, to find the RV we're looking for.

Reply:
true. Tbh I just wrapped the code there into a function and did not think about simplifying it, but you're right. Changed it.